### PR TITLE
Use Strict. Expose all the problems.

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Hey, y'all, don't accept this PR, it breaks things. Actually, it exposes things that are already broken. 

I've wrapped the whole thing in a closure and set it to use strict. Then, because some functions need to be in the global scope, I changed them from function statements to function expressions and attached them to window. Other than those changes, this file hasn't been changed.

Now go ahead and try to use it. You'll need to be in debug mode (so you get the uncompressed file) and then go to the admin and try to do some admin stuff. For example, in any list view you can try:
- Use the 'select all' checkbox
- Use a single-row checkbox
- Publish or unpublish items

You'll probably want to have your browser's console open while you do this stuff. You'll likely notice that:

a) it doesn't work
b) the console shows some errors

This is here to start a discussion but I've also got most of the fixes already done which I'll be submitting later.

### CODE REVIEW

**Comment by wilsonge** [comment]
I have something here that deals with the semi-colons https://github.com/joomla/joomla-cms/pull/6106

**Comment by okonomiyaki3000** [comment]
That's good. There are also several cases of undeclared variables. 

Beyond what's exposed by 'use strict', most of this code violates Joomla's js style rules. Also, a lot of it could simply be better written.

**Comment by okonomiyaki3000** [comment]
Oh, and line 37 is just plain wrong.

**Comment by dgt41** [comment]
That line came from #3484, but do you say it’s wrong?

**Comment by okonomiyaki3000** [comment]
Somehow it must have gotten changed since then. Look again.

**Comment by wilsonge** [comment]
Michaels fixed line 37 directly in core!

**Comment by Fedik** [comment]
@okonomiyaki3000  there some more error that need to fix after that, please be aware  :wink: 
you can see some, and copy from my another pull (#6357) https://github.com/Fedik/joomla-cms/commit/f670e8672c546a7f18e486cfe51989cf4cddfe40#diff-c1b1173fc1376e9afd63e8ec801c711aR102

File: media/system/js/core-uncompressed.js [hunk 1]
```diff
@@ -458,4 +462,6 @@ function checkAll_button(n, task) {
         }
     }
     submitform(task);
-}
+};
+
+}(Joomla));
```
Comment (Fedik) [inline media/system/js/core-uncompressed.js]: also you can add here `(window, document, Joomla)` and on the top also,
then js minifier will minify all `window, document, Joomla` variables in the script body :wink:

**Comment by okonomiyaki3000** [comment]
@Fedik this PR fixes nothing and is not meant to fix anything. It only exposes some of what needs to be fixed. Your points are correct but I won't do anything about them here.

**Comment by Kubik-Rubik** [comment]
#6512 merged, this PR can be closed. Thanks @okonomiyaki3000!

**Pull request closed without merging**